sn_interface/src/messaging/system/proposal.rs

```diff
@@ -40,20 +47,45 @@ impl Proposal {
         public_key_set: bls::PublicKeySet,
         index: usize,
         secret_key_share: &bls::SecretKeyShare,
-    ) -> Result<SectionSigShare> {
-        Ok(SectionSigShare::new(
-            public_key_set,
-            index,
-            secret_key_share,
-            &self.as_signable_bytes()?,
-        ))
+    ) -> Result<Either<SectionSigShare, (SectionSigShare, SectionSigShare)>> {
+        match &self.as_signable_bytes()? {
+            Either::Left(bytes) => Ok(Either::Left(SectionSigShare::new(
+                public_key_set,
+                index,
+                secret_key_share,
+                bytes,
+            ))),
+            Either::Right((bytes_1, bytes_2)) => {
+                let sig_1 =
+                    SectionSigShare::new(public_key_set.clone(), index, secret_key_share, bytes_1);
+                let sig_2 = SectionSigShare::new(public_key_set, index, secret_key_share, bytes_2);
+                Ok(Either::Right((sig_1, sig_2)))
+            }
+        }
     }
 
-    pub fn as_signable_bytes(&self) -> Result<Vec<u8>> {
+    #[allow(clippy::type_complexity)]
+    pub fn as_signable_bytes(&self) -> Result<Either<Vec<u8>, (Vec<u8>, Vec<u8>)>> {
         let bytes = match self {
             Self::VoteNodeOffline(node_state) => bincode::serialize(node_state),
             Self::RequestHandover(sap) => bincode::serialize(sap),
-            Self::NewElders(info) => bincode::serialize(&info.sig.public_key), // the pub key of the new elders
+            Self::NewElders(signed_sap) => bincode::serialize(&signed_sap.sig.public_key), // the pub key of the new elders
+            Self::NewSections { sap1, sap2 } => {
+                // the pub key of the new elders
+                let sap1_sig = bincode::serialize(&sap1.sig.public_key).map_err(|err| {
+                    Error::Serialisation(format!(
+                        "Couldn't serialise the Proposal '{:?}': {:?}",
+                        self, err
+                    ))
+                })?;
+                let sap2_sig = bincode::serialize(&sap2.sig.public_key).map_err(|err| {
+                    Error::Serialisation(format!(
+                        "Couldn't serialise the Proposal '{:?}': {:?}",
+                        self, err
+                    ))
+                })?;
```

Comment:
Does the error string really add much? I personally prefer clearer logic here
```suggestion
                let sap1_sig = bincode::serialize(&sap1.sig.public_key)?;
                let sap2_sig = bincode::serialize(&sap2.sig.public_key)?;
```

Reply:
This is the error that the code maps bincode serialise error to (see line 60 below, as it does not map the bincode error automatically here.

I just reused the same error from line 60.

If we want to modify/update the errors, that's a bit more than the scope of this refactor I'd say.

This PR intends to change nothing else in logic/handling/results, only get the siblings to be aware of eachother.

Reply:
Yeah, I went with that and reduced the number of declarations back to one again.